ui/src/app/data-browser/services/achilles.service.ts

```diff
@@ -0,0 +1,321 @@
+import { Analysis, AnalysisResult, IAnalysis} from '../AnalysisClasses';
+import { AnalysisDist, AnalysisDistResult } from '../AnalysisSubClasses';
+import { Concept, IConcept } from '../ConceptClasses';
+import { DomainClass } from '../DomainClasses';
+
+@Injectable()
+export class AchillesService {
+  constructor(
+     private api: DataBrowserService
+  ) { }
+  private handleError(error: any): Promise<any> {
+    console.error('An error occurred: ', error);
+    return Promise.reject(error.message || error);
+  }
+
+    // Return an Analysis object with results for the concepts array.
+    // This analysis obj can then be passed to app-chart
+    makeConceptsCountAnalysis(concepts: IConcept[]): Analysis {
+        const obj = {
+            analysisId: 3000,
+            analysisName: 'Number of Participants',
+            results: [],
+            chartType: 'column',
+            dataType: 'counts',
+            stratum1Name: ['Concept id']
+        };
+        for (let i = 0; i < concepts.length; i++) {
+            const ar =  new AnalysisResult({
+                analysisId: 3000,
+                stratum1Name: concepts[i].concept_name,
+                stratum1: concepts[i].concept_id,
+                countValue: concepts[i].count_value
+            });
+            obj.results.push(ar);
+        }
+        const a =  new Analysis(obj);
+        return a;
+    }
+
+  getAnalysisResults(a: IAnalysis): Promise<any[]> {
+
+      return this.api.getAnalysisResults(a.analysis_id, a.stratum[0], a.stratum[1],
+          a.stratum[2], a.stratum[3], a.stratum[4], a.stratum[5])
+          .toPromise()
+          .then(response => {
+            const data = response.items;
+            if (!data.length) {
+                return [];
+            }
+            let results =  data.map(item => {
+              let ar = null;
+              if (a.dataType === 'distribution') {
+                ar = new AnalysisDistResult(item);
+                return ar;
+              }
+              if (a.chartType === 'map') {
+                ar = new AnalysisResult(item);
+              } else {
+                  ar = new AnalysisResult(item);
+              }
+              return ar;
+            });
+
+            // If age decile results , arrange them in order filling in any missing deciles
+            if (a.analysis_id === 3102) {
+              results = Analysis.arrangeAgeDecileResults(results);
+            }
+            return results;
+
+          })
+          .catch(this.handleError);
+  }
+
+
+  /* Get all the analysis Objects we want to run for an analysisSection
+     If justThese is an array , just return ones with id in array  */
+  getSectionAnalyses( aids: number[]): Promise<IAnalysis[]> {
+    return this.api.getAnalyses()
+      .toPromise()
+      .then(response => {
+
+        let data = response.items;
+        // If justThese array analysis_ids then ,  filter list to those analysis_ids
```

Comment:
If there's filtering going on here, it seems like it should happen server-side...

Reply:
I could add array of parameters but in a hurry :) its a small list.